samples/cpp/gstreamer_pipeline.cpp

```diff
@@ -4,444 +4,373 @@
+            while (!cap->grab() && --limit != 0)
             {
-                cout << "Unsupported default codec: " << codec << endl;
-                cmd_parser->printErrors();
-                return string();
+                cout << "Skipping empty input frame - " << limit << endl;
             }
+            cap->retrieve(element);
+            tick.start();
+            *wrt << element;
+            tick.stop();
         }
-        else
-        {
-            cout << "Unsupported mode: " << resolution << endl;
-            cmd_parser->printErrors();
-            return string();
-        }
-        return plugin.str();
-    }
-    // Get fourcc for codec
-    int getFourccCode()
-    {
-        if      (codec == "h264")  { return VideoWriter::fourcc('H','2','6','4'); }
-        else if (codec == "h265")  { return VideoWriter::fourcc('H','E','V','C'); }
-        else if (codec == "mpeg2") { return VideoWriter::fourcc('M','P','E','G'); }
-        else if (codec == "mpeg4") { return VideoWriter::fourcc('M','P','4','2'); }
-        else if (codec == "mjpeg") { return VideoWriter::fourcc('M','J','P','G'); }
-        else if (codec == "vp8")   { return VideoWriter::fourcc('V','P','8','0'); }
-        else
+
+        if (fast_measure && tick.getCounter() >= 1000)
         {
-            cout << "Unsupported ffmpeg codec: " << codec << endl;
-            cmd_parser->printErrors();
-            return 0;
+            cout << "Fast mode frame limit reached - break" << endl;
+            break;
         }
-    }
-
-    // Check bad configuration
-    int checkConfiguration()
-    {
-        if ((codec == "mpeg2" && getGstMuxPlugin() == "qtmux")  ||
-            (codec == "h265"  && getGstMuxPlugin() == "avimux") ||
-            (pipeline == "gst-libav" && (codec == "h264" || codec == "h265"))   ||
-            (pipeline == "gst-vaapi1710" && codec=="mpeg2" && resolution=="4k") ||
-            (pipeline == "gst-vaapi1710" && codec=="mpeg2" && resolution=="1080p" && fix_fps > 30))
+        if (mode == "encode" && tick.getCounter() >= 1000)
         {
-            cout << "Unsupported configuration" << endl;
-            cmd_parser->printErrors();
-            return -1;
+            cout << "Encode frame limit reached - break" << endl;
+            break;
         }
-        return 0;
     }
-
-    bool   fast_measure;     // fast measure fps
-    string pipeline,         // gstreamer pipeline type
-           container,        // container type
-           mode,             // coding mode
-           codec,            // codec type
-           file_name,        // path to videofile
-           resolution;       // video resolution
-    int    fix_fps;          // fixed frame per second
-    Size   fix_size;         // fixed frame size
-    VideoWriter  wrt;
-    VideoCapture cap;
-    ostringstream stream_pipeline;
-    CommandLineParser* cmd_parser;
-};
-
-int main(int argc, char *argv[])
-{
-    try
+    total.stop();
```

Comment:
Before stop it is better to release capture and/or writer to cancel/flush pending operations.
Without that measurement is not accurate.